### PR TITLE
User profile updates

### PR DESCRIPTION
**Pull request by rtibbles**
## Summary
* Removes references to user_auth routes from user_profile plugin
* Uses a bare redirect instead
* Removes explicit reference to learn plugin in user_profile plugin, relies on next param instead

## References
Follow up to #8443

## Reviewer guidance
Does going to the profile page when not logged in redirect you to login?
Do you get properly redirected when prompted to fill out your demographic data?

----

## Testing checklist

- [ ] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [ ] Critical and brittle code paths are covered by unit tests

## PR process

- [ ] PR has the correct target branch and milestone
- [ ] PR has 'needs review' or 'work-in-progress' label
- [ ] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

## Reviewer checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

### CODE REVIEW

File: kolibri/plugins/user_profile/assets/src/routes.js [hunk 1]
```diff
@@ -1,6 +1,5 @@
 import store from 'kolibri.coreVue.vuex.store';
-import router from 'kolibri.coreVue.router';
-import { ComponentMap } from './constants';
+import { redirectBrowser } from 'kolibri.utils.redirectBrowser';
```
Comment (indirectlylit) [inline kolibri/plugins/user_profile/assets/src/routes.js]:
```suggestion
import redirectBrowser from 'kolibri.utils.redirectBrowser';
```

File: kolibri/plugins/learn/assets/src/views/LearnIndex.vue [hunk 2]
```diff
@@ -387,7 +387,8 @@
       },
       handleSubmitUpdateYourProfileModal() {
         if (this.userProfilePluginUrl) {
-          const url = `${this.userProfilePluginUrl()}#/profile/edit?next_page=learn`;
+          const next = window.encodeURIComponent(window.location.href);
+          const url = `${this.userProfilePluginUrl()}#/profile/edit?next=${next}`;
```
Comment (indirectlylit) [inline kolibri/plugins/learn/assets/src/views/LearnIndex.vue]: The modal which triggers this handler is conditional on `v-if="profileNeedsUpdate"`

```vue
    <UpdateYourProfileModal
      v-if="profileNeedsUpdate"
```

If the profile may be turned off completely, the modal should never be displayed and this handler should never be called. Therefore we should probably update the entire modal to be conditional on that:

```vue
    <UpdateYourProfileModal
      v-if="profileNeedsUpdate && userProfilePluginUrl"
```

and remove the condition from the 'submit' handler:

```js
      handleSubmitUpdateYourProfileModal() {
        const next = window.encodeURIComponent(window.location.href);
        const url = `${this.userProfilePluginUrl()}#/profile/edit?next=${next}`;
        const redirect = () => {
          window.location.href = url;
        };
        this.$store
          .dispatch('deferProfileUpdates', this.demographicInfo)
          .then(redirect, redirect);
      },
```

**Review by indirectlylit — COMMENTED** [review]
> Does going to the profile page when not logged in redirect you to login?

Yes. 

Filed minor follow-up here for 'the future': https://github.com/learningequality/kolibri/issues/8584

> Do you get properly redirected when prompted to fill out your demographic data?

No, the 'next' param is lost upon entering the 'edit' section of the profile:

  

That said, if I remember correctly we only included this prompt to edit and subsequent redirect when demographics was first introduced. Now that it's been a couple releases, I wonder if we'd be better served simply deleting all this prompt & redirect functionality to reduce complexity and maintenance burden?

**Comment by rtibbles** [comment]
I would be very happy to just remove the prompt and redirect functionality, it does seem like a lot of additional complexity for not a huge benefit.

**Comment by indirectlylit** [comment]
would you like me to take over that work or do you want to just update this PR?

**Comment by rtibbles** [comment]
I can just update this PR with deletions, no worries!

**Comment by rtibbles** [comment]
Have removed the learner profile notification functionality.

cc @radinamatic note that this has removed a feature file for the covered functionality.

**Comment by radinamatic** [comment]
> Have removed the learner profile notification functionality.
> cc @radinamatic note that this has removed a feature file for the covered functionality.

@pcenov Let's remove the relevant scenarios from the TestQuality platform too.

File: kolibri/core/assets/src/core-app/apiSpec.js [hunk 3]
```diff
@@ -55,7 +55,7 @@ import * as serverClock from '../serverClock';
 import * as resources from '../api-resources';
 import * as i18n from '../utils/i18n';
 import * as browserInfo from '../utils/browserInfo';
-import { redirectBrowser } from '../utils/redirectBrowser';
+import redirectBrowser from '../utils/redirectBrowser';
```
Comment (indirectlylit) [inline kolibri/core/assets/src/core-app/apiSpec.js:58]: was this broken more broadly?

**Review by indirectlylit — APPROVED** [review]
Tested and still properly redirects from profile -> auth when not logged in.

Deleted code is debugged code!

**Pull request merged**